### PR TITLE
tests: Add test to cover empty UGID in userinfo and JSON unmarshaling

### PR DESCRIPTION
**Pull request by denisonbarbosa**
We were missing these tests before, so this just adds them.

UDENG-2538

### CODE REVIEW

File: internal/brokers/broker_test.go [hunk 1]
```diff
@@ -210,12 +210,13 @@ func TestIsAuthenticated(t *testing.T) {
...
-		"Adds default groups even if broker did not set them":    {sessionID: "IA_info_missing_groups"},
-		"No error when auth.Next and no data":                    {sessionID: "IA_next"},
-		"No error when broker returns userinfo with empty gecos": {sessionID: "IA_info_missing_gecos"},
+		"Successfully authenticate":                                        {sessionID: "success"},
+		"Successfully authenticate after cancelling first call":            {sessionID: "IA_second_call", secondCall: true},
+		"Denies authentication when broker times out":                      {sessionID: "IA_timeout"},
+		"Adds default groups even if broker did not set them":              {sessionID: "IA_info_missing_groups"},
+		"No error when auth.Next and no data":                              {sessionID: "IA_next"},
+		"No error when broker returns userinfo with empty gecos":           {sessionID: "IA_info_missing_gecos"},
+		"No error when broker returns userinfo with group with empty GUID": {sessionID: "IA_info_missing_guid"},
```
Comment (didrocks) [inline internal/brokers/broker_test.go:219]: is it missing empty or missing from what the broker return? I think the broker return nothing which is then unmarshall to empty, but from the broker perspective, we didn’t get anything, or am I wrong? (just on naming the test case)

**Review by didrocks — COMMENTED** [review]
one small comment/question!

**Comment by didrocks** [comment]
Also, see my comment on the Jira card itself:

Sounds good to me! However, we are from an external user perspective and I think that gives some description on the behavior itself, without knowing the internal implementation. This is why, we want tests (even if they exercise the same code path) for both cases. Let’s imagine yaml unmarshall changes its behavior for $reasons, then, we have tests protecting us :)

File: internal/brokers/broker_test.go (hunk 1 above)
Comment (denisonbarbosa) [reply internal/brokers/broker_test.go:219]: This test case is to cover the case where the broker returns a group with empty ugid (now I realized that I switched around the letters :sweat_smile:) as in `{"name": "group1", "ugid": ""}`
